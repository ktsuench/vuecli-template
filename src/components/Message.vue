<template>
  <div>{{ msg }}</div>
</template>

<script>
export default {
  name: 'message',
  props: {
    msg: {
      type: String,
      default: 'default message',
    },
  },
};
</script>
