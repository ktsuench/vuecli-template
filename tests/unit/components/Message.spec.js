import { shallowMount } from '@vue/test-utils';
import Message from '@/components/Message.vue';

describe('Message', () => {
  describe('props.msg properties', () => {
    const wrapper = shallowMount(Message, { propsData: { msg: '' } });
    const msgProps = wrapper.vm.$options.props.msg;

    it('prop type is string', () => {
      expect(msgProps.type).toBe(String);
    });

    it('prop has default and default is an empty string', () => {
      expect(msgProps.default).toBe('default message');
    });
  });

  it('renders props.msg when passed', () => {
    const msg = 'new message';
    const wrapper = shallowMount(Message, {
      propsData: { msg },
    });
    expect(wrapper.text()).toBe(msg);
  });

  it('renders default message if not passed a prop', () => {
    const defaultMessage = 'default message';
    const wrapper = shallowMount(Message);
    expect(wrapper.text()).toBe(defaultMessage);
  });

  it('matches snapshot', () => {
    const msg = 'new message';
    const wrapper = shallowMount(Message, {
      propsData: { msg },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
