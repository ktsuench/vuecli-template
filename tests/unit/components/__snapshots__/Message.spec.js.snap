// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message matches snapshot 1`] = `<div>new message</div>`;
